x/clp/abci_test.go

```diff
@@ -223,3 +223,202 @@ func TestBeginBlocker(t *testing.T) {
 		})
 	}
 }
+
+func TestBeginBlocker_Incremental(t *testing.T) {
+	type ExpectedStates []struct {
+		pool              types.Pool
+		SwapPriceNative   sdk.Dec
+		SwapPriceExternal sdk.Dec
+		pmtpRateParams    types.PmtpRateParams
+	}
+
+	testcases := []struct {
+		name                   string
+		createBalance          bool
+		createPool             bool
+		createLPs              bool
+		poolAsset              string
+		address                string
+		nativeBalance          sdk.Int
+		externalBalance        sdk.Int
+		nativeAssetAmount      sdk.Uint
+		externalAssetAmount    sdk.Uint
+		poolUnits              sdk.Uint
+		poolAssetPermissions   []tokenregistrytypes.Permission
+		nativeAssetPermissions []tokenregistrytypes.Permission
+		params                 types.Params
+		epoch                  types.PmtpEpoch
+		expectedStates         ExpectedStates
```

Comment:
Can we read all these values from a file , it might be easier to geenrate the test cases from a json file

Reply:
But this can be done later as a refactor as well .

Reply:
yep agree json or a csv for those repetitive values would make more sense as we want to extend the tests to a greater number of blocks,